jetty-core/jetty-util-ajax/src/main/java/org/eclipse/jetty/util/ajax/JSON.java

```diff
@@ -1661,4 +1662,173 @@ public void addJSON(Appendable buffer)
+        {
+            if (isCommaNeede)
+            {
+                sb.append(", ");
+            }
+
+            sb.append('\n');
+            putIndent(sb, ident, nest + 1);
+            sb.append(putQuote(entry.getKey()));
+            sb.append(": ");
+            putIndent(sb, ident, 0);
+
+            Object value = entry.getValue();
+            if (value instanceof Boolean)
+            {
+                sb.append(value);
+                isCommaNeede = true;
+                continue;
+            }
+            if (value instanceof Double)
+            {
+                sb.append(value);
+                isCommaNeede = true;
+                continue;
+            }
+            if (value instanceof Long)
+            {
+                sb.append(value);
+                isCommaNeede = true;
+                continue;
+            }
+            if (value instanceof String)
+            {
+                sb.append(putQuote(value));
+                isCommaNeede = true;
+                continue;
+            }
+            if (value instanceof HashMap<?, ?>)
+            {
+                HashMap<?, ?> valueMap = (HashMap<?, ?>)value;
+                if (valueMap.size() == 0)
+                {
+                    sb.append("{}");
+                    isCommaNeede = true;
+                    continue;
+                }
+                else
+                {
+                    String v = toString(value, ident, nest + 2);
+                    sb.append(v);
+                    putIndent(sb, ident, nest + 1);
+                }
+                sb.append('}');
+                isCommaNeede = true;
+                continue;
+            }
+
+            if (value instanceof Object[])
+            {
+                sb.append(parseArray((Object[])value));
+                isCommaNeede = true;
+                continue;
+            }
+
+            if (value instanceof Object)
+            {
+                sb.append(parseArray((Object[])value));
+                isCommaNeede = true;
+                continue;
+            }
+
+            sb.append(putQuote(value));
+            isCommaNeede = true;
+        }
+        
+        sb.append('\n');
+        // Zero value means it is the most outermost nest
+        if (nest == 0)
+        {
+            sb.append('}');
+        }
+        return sb.toString();
+    }
+    
+    private String parseArray(Object[] o)
+    {
+        ArrayList<Object> array = new ArrayList<>();
+        for (Object val : o)
+        {
+            if (val instanceof Long)
+            {
+                array.add((Long)val);
+            }
+            else if (val instanceof String)
+            {
+                array.add("\"" + (String)val + "\"");
+            }
+            else if (val instanceof Boolean)
+            {
+                array.add((Boolean)val);
+            }
+        }
+        return array.toString();
+    }
+    
+    private String putQuote(Object o)
```

Comment:
StringUtil has methods that do this... check if they are usable or not

Reply:
Jetty's StringUtil doesn't have such a method.

Why not just implement this using an `Appendable` and reusing the existing `JSON.quotedEscape(Appendable buffer, String input)` for this purpose instead?

Reply:
Colleagues, I took into account your opinion and now use `quotedEscape` method. Thank you for hint!